### PR TITLE
EVG-14682 Evergreen Sort known-issue after failed

### PR DESCRIPTION
**Pull request by malikchaya2**

### CODE REVIEW

File: model/task/task.go [hunk 1]
```diff
@@ -2884,23 +2884,29 @@ func addStatusColorSort(key string) bson.M {
 							},
 							"then": 1, // red
 						},
+						{
+							"case": bson.M{
+								"$in": []interface{}{"$" + key, []string{evergreen.TaskKnownIssue}},
+							},
+							"then": 2,
```
Comment (SupaJoon) [inline model/task/task.go:2891]: can you remind me how the number weight maps to a color? what color does 2 map to?

Comment (SupaJoon) [reply model/task/task.go:2891]: I found that the color mapping actually happens completely in spruce and the color comments actually describe spruce implementation.

File: model/task/task.go [hunk 2]
```diff
@@ -2884,23 +2884,29 @@ func addStatusColorSort(key string) bson.M {
 							},
 							"then": 1, // red
 						},
+						{
+							"case": bson.M{
+								"$in": []interface{}{"$" + key, []string{evergreen.TaskKnownIssue}},
```
Comment (SupaJoon) [inline model/task/task.go:2889]: evergreen.TaskKnownIssue seems to be undefined from the logs: https://evergreen.mongodb.com/lobster/evergreen/task/evergreen_ubuntu1604_test_graphql_patch_4071b3c5984744e3a71589c95445d89f6db446be_60ad1ca52a60ed4e293079bf_21_05_25_15_49_58/1/task#bookmarks=0%2C204&l=1&shareLine=178

**Review by SupaJoon — CHANGES_REQUESTED** [review]

File: model/task/task.go (hunk 2 above)
Comment (malikchaya2) [reply model/task/task.go:2889]: it relies on https://github.com/evergreen-ci/evergreen/pull/4673

File: model/task/task.go (hunk 1 above)
Comment (malikchaya2) [reply model/task/task.go:2891]: Yes, the number is just a priority

**Comment by khelif96** [comment]
can you merge in main?

**Review by khelif96 — APPROVED** [review]
LGTM when tests complete

**Pull request merged**